xbitinfo/xbitinfo.py

```diff
@@ -35,6 +35,59 @@ def get_user_input():
     return args
 
 
+def get_bit_coords(dtype_size):
+    """Get coordinates for bits assuming float dtypes."""
+    if dtype_size == 16:
+        coords = (
+            ["±"]
+            + [f"e{int(i)}" for i in range(1, 6)]
+            + [f"m{int(i-5)}" for i in range(6, 16)]
+        )
+    elif dtype_size == 32:
+        coords = (
+            ["±"]
+            + [f"e{int(i)}" for i in range(1, 9)]
+            + [f"m{int(i-8)}" for i in range(9, 32)]
+        )
+    elif dtype_size == 64:
+        coords = (
+            ["±"]
+            + [f"e{int(i)}" for i in range(1, 12)]
+            + [f"m{int(i-11)}" for i in range(12, 64)]
+        )
+    else:
+        raise ValueError(f"dtype of size {dtype_size} neither known nor implemented.")
+    return coords
+
+
+def dict_to_dataset(info_per_bit):
+    """Convert keepbits dictionary to dataset."""
+    dsb = xr.Dataset()
+    for v in info_per_bit.keys():
+        dtype_size = len(info_per_bit[v])
+        dim_name = f"bit{dtype_size}"
+        dsb[v] = xr.DataArray(
+            info_per_bit[v],
+            dims=[dim_name],
+            coords={dim_name: get_bit_coords(dtype_size)},
+            name=v,
+            attrs={"long_name": f"{v} bitwise information"},
+        ).astype("float64")
```

Comment:
Should we add variables attrs here? Units Helps for context.

Reply:
What would be the unit? For bits it would be `-`, right?

Reply:
Let's set units ="" and original_units=old

Reply:
I put units="1" now, because that is the CF convention. For implementing the `original_units` we need to adjust `get_bitinformation` to put also metadata into the `info_per_bit` dictionary, so that it can also be written to disk and reloaded.

Reply:
then lets not do it. (we could add the metadata outside this function but then with `label` set, the attrs could potentially not match anymore when using a different file.)